iceoryx_examples/automotive_soa/src/owl/kom/method_client.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "owl/kom/method_client.hpp"
+
+namespace owl
+{
+namespace kom
+{
+MethodClient::MethodClient(const ServiceIdentifier& service,
+                           const InstanceIdentifier& instance,
+                           const MethodIdentifier& method) noexcept
+    : m_client({service, instance, method})
+{
+    m_waitset.attachState(m_client, iox::popo::ClientState::HAS_RESPONSE).expect("Failed to attach client!");
+}
+
+MethodClient::~MethodClient() noexcept
+{
+    m_waitset.detachState(m_client, iox::popo::ClientState::HAS_RESPONSE);
+
+    // Wait here if a thread or callback is still running
+    while (m_threadsRunning > 0)
+    {
+    }
+    std::lock_guard<iox::posix::mutex> guard(m_mutex);
```

Comment:
What does this mutex guard? I think it is not needed anymore.

Reply:
See my [previous answer](https://github.com/eclipse-iceoryx/iceoryx/pull/1355#discussion_r881670395). It protects several objects from being access at the very same time if more than one worker thread is still running.